File: leetcode/string/917. Reverse Only Letters/917. Reverse Only Letters.cpp
class Solution {
public:
    string reverseOnlyLetters(string s) {
        int j = s.size() - 1;
        string ans = s;

        for (int i = 0; i < s.size(); i++) {
            if (isalpha(s[i])) {
                while (!isalpha(s[j])) j--;
                ans[i] = s[j--];
            }
        }

        return ans;
    }
};

File: leetcode/string/917. Reverse Only Letters/917. Reverse Only Letters.rs
impl Solution {
    pub fn reverse_only_letters(s: String) -> String {
        let mut alphas: Vec<char> = s.chars().filter(|&c| c.is_ascii_alphabetic()).collect();
        let mut iter = alphas.into_iter().rev();

        s.chars().map(|c| {
            if !c.is_ascii_alphabetic() {
                c
            } else {
                iter.next().unwrap()
            }
        }).collect()
    }
}